x-pack/filebeat/input/httpjson/request.go

```diff
@@ -285,27 +290,41 @@ func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, p
 		urlCopy                 url.URL
 		urlString               string
 		httpResp                *http.Response
-		initialResponse         []*http.Response
-		intermediateResps       []*http.Response
-		finalResps              []*http.Response
 		isChainWithPageExpected bool
 		chainIndex              int
 	)
+	initialResponseChannel := make(chan *http.Response, 1)
+	defer func(n *int) {
+		r.log.Infof("request finished: %d events published", *n)
+	}(&n)
```

Comment:
```suggestion
	defer func() {
		r.log.Infof("request finished: %d events published", n)
	}()
```
The deferred func is a closure, so it is already holding `n` as a reference to the enclosing scope's var `n`.

Reply:
Got it. I'll update it.